### PR TITLE
parity with UI

### PR DESCRIPTION
**Pull request by devidask27**
## ISSUE(S):

https://platform9.atlassian.net/browse/FT-337

## SUMMARY

Currently, the UI provides some more feature options which are not present in the CLI bootstrap command. The CLI should have parity in terms of supported flags with UI.

## ISSUE TYPE

- [] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that may cause existing functionality to not work as expected)
- [] This change requires a documentation update

## IMPACTED FEATURES/COMPONENTS:

Impacted commands
bootstrap

#### Manual

**Current bootstrap cluster**
Screenshot 2022-02-25 at 12 38 19 PM
**bootstrap cluster after vs UI cluster**
Screenshot 2022-02-25 at 12 21 15 PM

Screenshot 2022-02-25 at 12 34 13 PM
Screenshot 2022-02-25 at 12 21 34 PM

**Incorrect pmk version**

Screenshot 2022-03-04 at 11 58 32 AM

#### Reviewers

/cc @anupbarve @nikhilbandi

### CODE REVIEW

File: pkg/qbert/qbert.go [hunk 1]
```diff
@@ -419,6 +473,54 @@ func Attach_Status(attachEndpoint string, token string, byt []byte) (*http.Respo
 	return resp, nil
 }
 
+func updatePayload(p string) string {
+	var monitoringDetails string
+	var tagDetails string
+	if IsPMKversionDefined {
+		if SplitPMKversion[0] <= "1.20.11" {
```
Comment (anupbarve) [inline pkg/qbert/qbert.go:480]: Put the version in constants.go
Also document the logic as to why we are having such a condition wrt the version.

File: pkg/qbert/qbert.go [hunk 2]
```diff
@@ -419,6 +473,54 @@ func Attach_Status(attachEndpoint string, token string, byt []byte) (*http.Respo
...
+		monitoringDetails = getMonitoringDetails()
+	}
+	p = p[:len(p)-1]
+	p = p + tagDetails + monitoringDetails + "}"
+	return p
+}
+
+func getMonitoringDetails() string {
+	if !IsMonitoringDisabled {
+		Monitoring = fmt.Sprintf(`,"monitoring":"{\"retentionTime\":\"7d\"}"`)
```
Comment (anupbarve) [inline pkg/qbert/qbert.go:495]: Why 7 days ? 
Please document the logic

File: cmd/bootstrap.go [hunk 3]
```diff
@@ -43,35 +43,79 @@ var bootstrapCmd = &cobra.Command{
...
+	masterVIP                string
+	masterVIPIf              string
+	metallbIPRange           string
+	containersCIDR           string
+	servicesCIDR             string
+	externalDNSName          string
+	privileged               bool
+	allowWorkloadsOnMaster   bool
+	networkPlugin            string
+	calicoNatOutgoing        int
```
Comment (anupbarve) [inline cmd/bootstrap.go:118]: All these variables need to have corresponding comments.

File: cmd/bootstrap.go [hunk 4]
```diff
@@ -80,6 +124,33 @@ func bootstrapCmdRun(cmd *cobra.Command, args []string) {
...
+	}
+
+	qbert.IsPMKversionDefined = cmd.Flags().Changed("pmk-version")
+	if qbert.IsPMKversionDefined {
+		//Profile Engine support check
+		qbert.SplitPMKversion = strings.Split(pmkVersion, "-")
+		if qbert.SplitPMKversion[0] < "1.20.11" {
+			containerRuntime = "docker"
+			enableProfileEngine = false
+		}
```
Comment (anupbarve) [inline cmd/bootstrap.go:146]: Move all the keywords to constants.go
Examples include, but not limited to :)
... 
calico
etcd-backup
docker
... etc

**Review by anupbarve — CHANGES_REQUESTED** [review]
Some generic comments before this code could be reviewed.

- Please update README.md
- Please use the bug template to raise this PR in order to give more info about this change.
- Do we need to add test cases for these changes ?
- I hope you have got all the command parameters reviewed from Anmol.

**Comment by devidask27** [comment]
> Some generic comments before this code could be reviewed.
> 
> * Please update README.md
> * Please use the bug template to raise this PR in order to give more info about this change.
> * Do we need to add test cases for these changes ?
> * I hope you have got all the command parameters reviewed from Anmol.

Updated README.md
Added PR template
I think no need of test cases, added two functions but it only contains fmt.Sprintf functionality
Already given the binary to anmol to try it

File: README.md [hunk 5]
```diff
@@ -590,24 +590,45 @@ Usage:
...
+  -r, --remove-existing-pkgs                Will remove previous installation if found (default false)
+      --reserved-cpu string                 Comma separated list of CPUs to be reserved for the system, e.g: 4-8,9-12
+      --scheduler-flags strings             Comma separated list of supported Kube-scheduler flags, e.g: --kube-api-burst=120,--log_file_max_size=3000
+      --services-cidr string                CIDR for services overlay (default "10.21.0.0/16")
+  -s, --ssh-key string                      Ssh key file for connecting to the node
+  -e, --sudo-pass string                    Sudo password for user on remote host
+      --tag string                          Add tag metadata to this cluster (key=value)
+      --topology-manager-policy string      Topology manager policy (default "none")
+      --use-hostname                        Use node hostname for cluster creation
+  -u, --user string                         Ssh username for the node
```
Comment (anupbarve) [inline README.md:631]: IMHO, user is going to get overwhelmed after looking at this list of options. 
We should provide examples of how to get a cluster bootstrapped with default options.
These examples should also be part of the usage.

**Comment by devidask27** [comment]
@anupbarve Tried to Resolve points you mentioned

- groped flags as optional and required
- added bootstrap command example in command usage message
Screenshot 2022-03-14 at 12 52 48 PM

File: README.md [hunk 6]
```diff
@@ -587,9 +587,11 @@ Node decommissioning started....This may take a few minutes....Check the latest
 Bootstrap a single node Kubernetes cluster with current node as the master node.
 
 Usage:
-  pf9ctl bootstrap [flags] cluster-name
+  pf9ctl bootstrap [flags] cluster-name, e.g: pf9ctl bootstrap testCluster --pmk-version 1.21.3-pmk.72
```
Comment (anupbarve) [inline README.md:590]: These versions keep getting changed. How would the user know which version to use ?

Comment (devidask27) [reply README.md:590]: we are fetching all versions first, if user have not passed any version or he entered wrong version then we are listing all versions and asking to select one. 
we can remove this hardcoded version and can make it `pf9ctl bootstrap testCluster --pmk-version <version>`

Comment (anupbarve) [reply README.md:590]: yes, please remove the hardcoded version string from here and below as well.

**Review by anupbarve — CHANGES_REQUESTED** [review]

File: README.md (hunk 6 above)
Comment (devidask27) [reply README.md]: Done

**Review by anupbarve — APPROVED** [review]

**Pull request merged**